PYME/Acquire/ui/actionUI.py

```diff
@@ -175,29 +177,30 @@ def OnAddSequence(self, event):
     
     def _add_ROIs(self, rois):
         """
-        Add ROIs to queue, staggering their positioning and spooling actions
+        Add ROI positioning and spooling actions to queue.
 
         Parameters
         ----------
         rois: list-like
-            list of ROI (x, y) positions, or array of shape (n_roi, 2)
-
-        Returns
-        -------
+            list of ROI (x, y) positions, or array of shape (n_roi, 2). Units in micrometers.
 
         """
         
-        nice = float(self.tNice.GetValue())
-        timeout = float(self.tTimeout.GetValue()) #CHECKME - default here might be too short
-        
         # coordinates are for the centre of ROI, and are referenced to the 0,0 pixel of the camera,
         # correct this for a custom ROI.
         roi_offset_x, roi_offset_y = self.scope.get_roi_offset()
 
-        for x, y in rois:
-            args = {'state': {'Positioning.x': float(x) - roi_offset_x, 'Positioning.y': float(y) - roi_offset_y}}
+        # do a greedy sort relative to our stage position
+        positions = np.reshape(rois, (len(rois), 2)).astype(float) - np.array([roi_offset_x, roi_offset_y])[None, :]
+
+        n_frames = int(self.tNumFrames.GetValue())
+        nice = float(self.tNice.GetValue())
+        time_per_task = n_frames * n_frames / self.scope.cam.GetFPS()
+        timeout = max(float(self.tTimeout.GetValue()), 1.25 * time_per_task)  # allow enough time for what we queue
```

Comment:
I'm pretty sure this timeout calculation is not doing what you want - needs to be > ` n_sequences_queued*n_frames/FPS` (but even this is not particularly safe if there are any higher priority tasks in there)